### PR TITLE
chore(switch): Implementation of the fluid switch component.

### LINKED ISSUE #1222: Switch as webcomponent
# Feature Request

## Summary

We want to have a switch component for next as a webcomponent. 

## Feature Description

The switch should look like 
image

Design details are available in the Figma file from UX. If you need a link - please reach out. 

Additional important information and tips could be found here - https://inclusive-components.design/toggle-button/
and here https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-1/checkbox-1.html

Scope of this issue is to create a first basic version that uses our design tokens including a first set of tests. Check out the checkbox for guidance.
If a11y is a big topic - please create followup issues to include these into the component. 

Animations can be omitted and added in a separate PR/issue. 

Check out https://www.fast.design/docs/components/switch as a reference as well.

### PR DESCRIPTION
**Pull request by bardosmisi**
Co-authored-by: Rowa Audil <rowa.audil@dynatrace.com>

### <strong>Pull Request</strong>

Fluid Switch Component.

#### Type of PR

Feature (non-breaking change which adds functionality)

Fixes #1222

### CODE REVIEW

File: libs/fluid-elements/switch/src/lib/switch.ts [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+  css,
+} from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map';
+import { FLUID_SPACING_3X_SMALL } from '@dynatrace/fluid-design-tokens';
+
+export class FluidSwitchChangeEvent extends CustomEvent<any> {
+  constructor(public checked: boolean) {
+    super('change', { bubbles: true, composed: true });
+  }
+}
```
Comment (tomheller) [inline libs/fluid-elements/switch/src/lib/switch.ts:33]: Could be moved out into a separete file.

File: libs/fluid-elements/switch/src/lib/switch.ts [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  LitElement,
+  TemplateResult,
+  html,
+  property,
+  CSSResult,
+  unsafeCSS,
+  css,
+} from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map';
+import { FLUID_SPACING_3X_SMALL } from '@dynatrace/fluid-design-tokens';
```
Comment (tomheller) [inline libs/fluid-elements/switch/src/lib/switch.ts:27]: Will work when #1291 is done.

File: libs/fluid-elements/switch/src/lib/switch.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+      .fluid-svg-switch:hover .fluid-switch-glow {
+        opacity: 0.1;
+      }
+      .fluid-switch-glow {
+        opacity: 0;
+        fill: var(--fluid-switch--focus);
+      }
+      .fluid-svg-switch .fluid-switch-glow {
+        transition: transform 0.3s;
+        /* transform: translateX(16px); */
```
Comment (tomheller) [inline libs/fluid-elements/switch/src/lib/switch.ts:142]:
```suggestion
```

**Comment by tomheller** [comment]
If you do not want to be blocked by #1291 you could move the style back to the scss file (no need to change the selectors / structure. css is valid scss), and replace the token imports with sass ones. 

I could update 1291 with the changes from the switch later on as well!

**Comment by rowa-audil** [comment]
@tomheller That would be awesome. Will jump on it right now.

**Pull request merged**